### PR TITLE
QH - DSPDC-763 - Parameterized the ENCODE extraction workflow + Added integration test

### PR DESCRIPTION
**Pull request by quazi-broad**
The Encode extraction workflow can now take in query parameters.
This will allow us to pass more targeted entry-points to the pipeline during testing and also allow us to query for other organism types (currently hardcoded to pull human donor data).
I also added an integration test that just takes one biosample accession and makes sure that we downloaded some stuff to the temporary directories. Not sure how often biosamples change - it would be great to use a biosample we know will never change. Maybe Encode has some seed data for this purpose?

### CODE REVIEW

File: extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  val AnalysisStepVersionPath = (outputDir.pathAsString + "/AnalysisStepVersion")
+  val AntibodyLotPath = (outputDir.pathAsString + "/AntibodyLot")
+  val BiosamplePath = (outputDir.pathAsString + "/Biosample")
+  val DonorPath = (outputDir.pathAsString + "/Donor")
+  val ExperimentPath = (outputDir.pathAsString + "/Experiment")
+  val FilePath = (outputDir.pathAsString + "/File")
+  val FunctionalCharacterizationExperimentPath = (outputDir.pathAsString + "/FunctionalCharacterizationExperiment")
+  val LibraryPath = (outputDir.pathAsString + "/Library")
+  val ReplicatePath = (outputDir.pathAsString + "/Replicate")
+  val TargetPath = (outputDir.pathAsString + "/Target")
```
Comment (danxmoran) [inline extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala]: Style nitpick: it'd be more conventional for these to all be lower-camel-case, i.e.
```suggestion
  val targetPath = (outputDir.pathAsString + "/Target")
```
Upper-camel-case is typically reserved for static constants

File: extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  /// VERIFY DOWNLOADS
+  behavior of "ExtractionPipeline"
+
+  it should "successfully download results from the from Encode API" in {
+    outputDir.nonEmpty should be(true)
+    println("outputDir has contents...")
```
Comment (danxmoran) [inline extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala]: I'd drop all the `println`s, the normal test output should be sufficient

File: extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  val FunctionalCharacterizationExperimentPath = (outputDir.pathAsString + "/FunctionalCharacterizationExperiment")
+  val LibraryPath = (outputDir.pathAsString + "/Library")
+  val ReplicatePath = (outputDir.pathAsString + "/Replicate")
+  val TargetPath = (outputDir.pathAsString + "/Target")
+
+  /// VERIFY DOWNLOADS
+  behavior of "ExtractionPipeline"
+
+  it should "successfully download results from the from Encode API" in {
+    outputDir.nonEmpty should be(true)
```
Comment (danxmoran) [inline extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala]: Does this check add anything over the more specific checks below?

File: extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/Args.scala [hunk 4]
```diff
@@ -20,5 +22,22 @@ case class Args(
...
+  batchSize: Long,
+  @HelpMessage(
+    "Initial query to target a specific entry-point to the pipeline."
+  )
+  initialQuery: List[(String, String)] = List("organism.name" -> "human")
 )
+
+object Args {
+
+  implicit val customArgParser: ArgParser[(String, String)] =
```
Comment (danxmoran) [inline extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/Args.scala]: Now that it's working, would be more specific with the name
```suggestion
  implicit val tupleParser: ArgParser[(String, String)] =
```

File: extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineBuilder.scala [hunk 5]
```diff
@@ -67,7 +67,7 @@ class ExtractionPipelineBuilder(getClient: () => EncodeClient)
 
     // biosamples are the first one and follow a different pattern, so we don't use the generic method
     val biosamples = ctx
-      .parallelize(List(List("organism.name" -> "human")))
+      .parallelize(List({ args.initialQuery }))
```
Comment (danxmoran) [inline extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineBuilder.scala]: Brackets not needed here
```suggestion
      .parallelize(List(args.initialQuery))
```

File: extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala (hunk 3 above)
Comment (quazi-broad) [reply extraction/src/it/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineIntegrationSpec.scala]: No, I actually had written this first to make sure the whole test script was working before I decided to go down a level and check within the individual entity files. I'll remove this check.

**Pull request merged**